### ISSUE TITLE
KSH93 extended regular Expressions

### ISSUE THREAD

**Issue by jpolifka** [koalaman/shellcheck#1346]
#### For bugs
- Rule Id: SC1073
- My shellcheck version (`shellcheck --version` or "online"):  "online"
- [x ] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /usr/bin/ksh93
 if [[ ${vti_handle} != {9}([A-Z0-9]-){5}([A-Z0-9]-){5}([A-Z0-9]-){5}([A-Z0-9]-){12}([A-Z0-9]) ]];then
   echo "ERR - VTI - VTI Job-Handle nicht korrekt. Wert: ${vti_handle}"
   return 5
 else
   echo "DBG - VTI - Job ${vti_job} gestartet."
   echo ${vti_handle}
 fi

```

#### Here's what shellcheck currently says:
SC1009: The mentioned syntax error was in this if expression.
SC1073: Couldn't parse this test expression. Fix to allow more checks.
SC1083: This { is literal. Check expression (missing ;/\n?) or quote it.
SC1083: This } is literal. Check expression (missing ;/\n?) or quote it.
SC1036: '(' is invalid here. Did you forget to escape it?
SC1072: Expected test to end here (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:
We are using KSH93 on AIX.
The above extended regular Expression checks for:
* [A-Z0-9]- with 9 occurrences
* [A-Z0-9]- with 5 occurrences
* [A-Z0-9]- with 5 occurrences
* [A-Z0-9]- with 5 occurrences
* [A-Z0-9] with 12 occurrences

In KSH93 the interval qualifiers starts the regexp, see here:
https://blog.fpmurphy.com/2009/01/ksh93-regular-expressions.html

Is there any chance, that shellcheck can match these regular expressions?
